dice_ml/counterfactual_explanations.py

```diff
@@ -127,11 +92,134 @@ def visualize_as_list(self, display_sparse_df=True,
+
+            local_importance_matrix = None
+            if self.local_importance is not None:
+                local_importance_matrix = []
+                for local_importance_dict in self.local_importance:
+                    local_importance_list = []
+                    for feature_name in feature_names:
+                        local_importance_list.append(local_importance_dict.get(feature_name))
+                    local_importance_matrix.append(local_importance_list)
+
+            summary_importance_list = None
+            if self.summary_importance is not None:
+                summary_importance_list = []
+                for feature_name in feature_names:
+                    summary_importance_list.append(self.summary_importance.get(feature_name))
+
+            entire_dict = {
+                'test_data': combined_test_instance_list,
+                'cfs_list': combined_final_cfs_list,
+                'local_importance': local_importance_matrix,
+                'summary_importance': summary_importance_list,
+                'data_interface': data_interface,
+                'feature_names': feature_names,
+                'feature_names_including_target': feature_names_including_target,
+                'model_type': model_type,
+                'desired_class': desired_class,
+                'desired_range': desired_range,
+                'metadata': self.metadata
+            }
+            return json.dumps(entire_dict)
+        else:
+            raise Exception("Unsupported serialization version {}".format(
+                serialization_version))
 
     @staticmethod
     def from_json(json_str):
         """ Deserialize json string to a CounterfactualExplanations object.
         """
-        return json.loads(json_str, object_hook=as_counterfactual_explanations)
+        json_dict = json.loads(json_str)
+        if 'metadata' in json_dict:
+            version = json_dict['metadata'].get('version')
+            if version is None:
+                raise UserConfigValidationException("No version field in the json input")
+            elif not _check_supported_json_output_versions(version):
+                raise UserConfigValidationException("Incompatible version {} found in json input".format(version))
+
+            if version == '1.0':
+                cf_examples_list = []
+                for cf_examples_str in json_dict["cf_examples_list"]:
+                    cf_examples_list.append(CounterfactualExamples.from_json(cf_examples_str))
+
+                return CounterfactualExplanations(
+                        cf_examples_list=cf_examples_list,
+                        local_importance=json_dict["local_importance"],
+                        summary_importance=json_dict["summary_importance"])
+            elif version == '2.0':
+                cf_examples_list = []
+                for index in range(0, len(json_dict['cfs_list'])):
+                    cf_examples_str = json.dumps(
```

Comment:
same question here. We do load the string but then again create a string using `dumps`. Perhaps it will be good to add an inline comment on why these transformations are needed.

Reply:
Again during deserialization we need to recompose the necessary components for CounterfactualExamples and call CounterfactualExamples' from_json() method which expects a json string.

Reply:
I will document this.